### DISCUSSION TITLE
Limit frequency connection

### DISCUSSION THREAD

**Discussion by xrivo1** [rejetto/hfs#185] [Ideas]
I found in the log, some IP addresses that in just 1 minute generated 130 lines. The situation is similar to a DDoS attack. It would be useful to insert a function that automatically bans addresses that simulate such attacks.

**Comment by rejetto** [comment]
Ddos are "distributed", IPs are many, and you will probably not be able to block them.
Anyway, I made a simple antidos (not distributed) plugin, in case the person bothering you is doing it from one computer.
Find it in Plugins/Search

<img width="341" alt="image" src="https://user-images.githubusercontent.com/1367199/229456277-05666667-73ac-4718-86a9-3fed890dc020.png">

**Reply by xrivo1** [reply]
Perfect. installed and everything works. just one question..... Does the autoban automatically add the IP to the list with those entered manually?
Thank you

**Comment by rejetto** [comment]
nope, it's a separated list, and it's not even saved on disk, that's why it is reset when you restart hfs (or the plugin)

**Comment by xrivo1** [comment]
Ciao. Ho settato l'antiDDoS con valori limiti. 50 richieste entro 1 minuto.
Oggi ho trovato un IP che ha eseguito 97 richieste in 5 secondi e non è stato bloccato.
Ho sbagliato io qualcosa nel settaggio?

**Reply by rejetto** [reply]
e tu delle 97 richieste lo sai dai log? vuoi farmi vedere?

**Comment by xrivo1** [comment]
eccolo
[access-error.log](https://github.com/rejetto/hfs/files/11190733/access-error.log)

**Reply by rejetto** [reply]
e sei sicuro che il plugin stesse girando?

**Reply by rejetto** [reply]
c'è disponibile un aggiornamento, forse ho fixato.

**Reply by xrivo1** [reply]
Si. il plugin girava. 
Provo ad aggiornarlo.

**Reply by xrivo1** [reply]
cavolo. me lo vede come trojan grave......... mi annulla il download

**Reply by rejetto** [reply]
ma scusa in che senso? l'aggiornmento intendevo del plugin, non di HFS.
Admin > plugins > check updates

**Reply by rejetto** [reply]
che antivirus è?
quindi tu ora non stai usando la 0.43.0 ?

**Reply by xrivo1** [reply]
quando aggiorno il plugin, l'antivirus di windows me lo blocca

**Reply by xrivo1** [reply]
![antivirus](https://user-images.githubusercontent.com/106228981/230949440-edc98029-34c7-4d90-83e5-f84159798849.jpg)

**Reply by rejetto** [reply]
oddio, siamo arrivati a questo!
comunque ho provato sul mio windows a installare (non aggiornare) antidos, e non ha fiatato.

<img width="361" alt="image" src="https://user-images.githubusercontent.com/1367199/230953629-18b60086-75e1-4a42-b948-151330826255.png">

Ho provato pure a stuzzicare Windows Defender e gli ho detto di analizzare quel file, e niente
<img width="352" alt="image" src="https://user-images.githubusercontent.com/1367199/230954843-dec3bcbb-5668-4f68-b544-69f371746d70.png">

non so proprio che dirti...

**Reply by xrivo1** [reply]
ho disinstallato e provato a reinstallare antidos. nulla. segnala sempre trojan. 
...... proverò a consentire l'esecuzione a mio rischio e pericolo :-)

**Reply by rejetto** [reply]
dopo che hai consentito il file puoi sottoporlo a https://www.virustotal.com/

**Reply by xrivo1** [reply]
ok. ho detto all'antivirus di accettare il file. ora è installato. ti farò sapere se funziona

**Comment by xrivo1** [comment]
Sei il migliore. Pare che l' antidos stanotte abbia fatto il suo lavoro.
Ti allego log e screenshots di powershel
[access-error.log](https://github.com/rejetto/hfs/files/11208907/access-error.log)
![antidos banning](https://user-images.githubusercontent.com/106228981/231397800-133edd4e-ee72-4ffd-8945-02beb67cbe28.jpg)
